opentracing/span.py

```diff
@@ -20,45 +20,83 @@
 
 from __future__ import absolute_import
 
-import re
+
+class SpanContext(object):
+    def set_baggage_item(self, key, value):
+        """Stores a Baggage item in the span as a key/value pair.
+
+        Enables powerful distributed context propagation functionality where
+        arbitrary application data can be carried along the full path of
+        request execution throughout the system.
+
+        Note 1: Baggage is only propagated to the future (recursive) children
+        of this SpanContext.
+
+        Note 2: Baggage is sent in-band with every subsequent local and remote
+        calls, so this feature must be used with care.
+
+        :param key: Baggage item key
+        :type key: str
+
+        :param value: Baggage item value
+        :type value: str
+
+        :rtype : SpanContext
+        :return: itself, for chaining the calls.
+        """
+        return self
+
+    def get_baggage_item(self, key):
+        """Retrieves value of the Baggage item with the given key.
+
+        :param key: key of the Baggage item
+        :type key: str
+
+        :rtype : str
+        :return: value of the Baggage item with given key, or None.
+        """
+        return None
 
 
 class Span(object):
     """
     Span represents a unit of work executed on behalf of a trace. Examples of
     spans include a remote procedure call, or a in-process method call to a
-    sub-component. A trace is required to have a single, top level "root"
-    span, and zero or more children spans, which in turns can have their own
-    children spans, thus forming a tree structure.
+    sub-component. Every span in a trace may have zero or more causal parents,
+    and these relationships transitively form a DAG. It is common for spans to
+    have at most one parent, and thus most traces are merely tree structures.
 
     Span implements a Context Manager API that allows the following usage:
 
     .. code-block:: python
 
-        span = tracer.start_span(operation_name='go_fishing')
-        with span:
+        with tracer.start_span(operation_name='go_fishing') as span:
             call_some_service()
 
-    In this case it's not necessary to call span.finish()
+    In the Context Manager syntax it's not necessary to call span.finish()
     """
 
     def __init__(self, tracer):
         self._tracer = tracer
```

Comment:
I think it's reasonable to take the `context` as constructor argument as well, so that implementations get it for free. I am not even sure if it's possible for subclasses to override `@property context`